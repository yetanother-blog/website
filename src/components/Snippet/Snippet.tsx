import styled from 'styled-components';

export const Snippet = styled.iframe.attrs(() => ({
  loading: 'lazy',
  sandbox: 'allow-scripts',
}))`
  width: 100%;
  margin-bottom: 0;
  border: 0;
`;
