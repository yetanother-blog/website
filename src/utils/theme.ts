import { DefaultTheme } from 'styled-components';

export const theme: DefaultTheme = {
  fontSizes: ['12px', '14px', '16px', '20px', '24px', '32px'],
  colors: {
    primary: '#00FF00',
    secondary: '#FB00FF',
    grey500: '#1C1C1C',
    grey400: '#ACADAC',
    grey300: '#F6F6F6',
  },
  breakpoints: ['576px', '768px', '992px', '1200px'],
  space: {
    zero: 0,
    xxs: 4,
    xs: 8,
    s: 12,
    m: 16,
    l: 20,
    xl: 40,
    xxl: 60,
    xxxl: 120,
  },
  radii: {
    s: 3,
  },
  width: {
    m: '620px',
    l: '940px',
    xl: '1200px',
  },
};
