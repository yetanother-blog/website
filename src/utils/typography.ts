import { CSSObject } from "styled-components";
import Typography from "typography";

const typography = new Typography({
  googleFonts: [
    {
      name: "Source Sans Pro",
      styles: ["400", "600"]
    },
    {
      name: "Source Code Pro",
      styles: ["600", "700"]
    }
  ],
  baseFontSize: "20px",
  baseLineHeight: 1.85,
  bodyFontFamily: ["Source Sans Pro", "sans-serif"],
  boldWeight: "600",
  overrideStyles: ({ adjustFontSizeTo }, options, styles) => ({
    a: {
      fontFamily: `"Source Code Pro", "sans-serif"`
    }
  })
});

// Hot reload typography in development.
if (process.env.NODE_ENV !== `production`) {
  typography.injectStyles();
}

export default typography;

export const rhythm = typography.rhythm;
export const scale = typography.scale;

type StyledScale = (values: number) => CSSObject;
export const styledScale = scale as StyledScale;
