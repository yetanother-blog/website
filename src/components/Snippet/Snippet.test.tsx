import React from 'react';
import { render } from '@testing-library/react';
import { Snippet } from './Snippet';
import { ThemeContext } from 'styled-components';
import { theme } from '../../utils/theme';

describe('Snippet component', () => {
  it('should render', () => {
    const { container } = render(
      <ThemeContext.Provider value={theme}>
        <Snippet
          url="https://example.com"
          title="Title is coooool :)"
          height={256}
        />
      </ThemeContext.Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
