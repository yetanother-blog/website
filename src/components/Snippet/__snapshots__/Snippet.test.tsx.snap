// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snippet component should render 1`] = `
<div>
  <iframe
    class="sc-bdvvaa fvMnWP"
    height="256"
    loading="lazy"
    src="data:text/html;charset=utf-8,%3Chtml%3E%3Cbody%3Efoo%3C/body%3E%3C/html%3E"
    title="Title is coooool :)"
  />
</div>
`;
